core/template.js

```diff
@@ -27,28 +24,28 @@
 	 * @param {String} source The template source.
 	 */
 	CKEDITOR.template = function( source ) {
-		// Builds an optimized function body for the output() method, focused on performance.
 		// For example, if we have this "source":
 		//	'<div style="{style}">{editorName}</div>'
 		// ... the resulting function body will be (apart from the "buffer" handling):
 		//	return [ '<div style="', data['style'] == undefined ? '{style}' : data['style'], '">', data['editorName'] == undefined ? '{editorName}' : data['editorName'], '</div>' ].join('');
 
 		// Try to read from the cache.
-		if ( cache[ source ] )
+		if ( cache[ source ] ) {
 			this.output = cache[ source ];
+		}
 		else {
-			var fn = source
-				// Escape chars like slash "\" or single quote "'".
-				.replace( reEscapableChars, '\\$1' )
-				.replace( reNewLine, '\\n' )
-				.replace( reCarriageReturn, '\\r' )
-				// Inject the template keys replacement.
-				.replace( rePlaceholder, function( m, key ) {
-					return "',data['" + key + "']==undefined?'{" + key + "}':data['" + key + "'],'";
+			// less performant solution when CSP is disabling indirect evaluation
+			cache[ source ] = function( data, buffer ) {
```

Comment:
Did some profiling against http://tests.ckeditor.dev:1030/#tests/is:unit,group:Core. There are more than 2000 tests so both functions were run probably around few thousand times. Not sure though how much reliable is profiling tool. Anyway (headers may not be visible soe first column is **Self Time** and the second one - **Total Time**):

### Current solution

**CKEDITOR.template**
![image](https://cloud.githubusercontent.com/assets/1061942/24490671/eba5e350-1523-11e7-987f-08c49f5accf4.png)


**CKEDITOR.output**
![image](https://cloud.githubusercontent.com/assets/1061942/24490783/7deff584-1524-11e7-8631-f0065540b31c.png)


### Simplified

**CKEDITOR.template** (not sure why there are two records for each)
![image](https://cloud.githubusercontent.com/assets/1061942/24490402/c0ed2f3e-1522-11e7-8d0d-1463e8083e67.png)
![image](https://cloud.githubusercontent.com/assets/1061942/24490431/d96bdc90-1522-11e7-8fdf-4c7ee1ef3cba.png)

**CKEDITOR.output** (also two records)
![image](https://cloud.githubusercontent.com/assets/1061942/24490494/1f1b9f3c-1523-11e7-9b0e-23bd0cc4bd2f.png)
![image](https://cloud.githubusercontent.com/assets/1061942/24490538/48e8a300-1523-11e7-9786-817d9c5fe52e.png)

So it seems execution time is very similar for both so maybe it will be better to follow simpler approach? Also looking at those numbers it seems the introduced change (in comparison to previous CSP unsafe one) should not have any negative impact on editor responsibility.

WDYT?